{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromsome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ensembl id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11868</td>\n",
       "      <td>14412</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14362</td>\n",
       "      <td>29806</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>WASH7P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29553</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>MIR1302-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34553</td>\n",
       "      <td>36081</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>FAM138A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52472</td>\n",
       "      <td>54936</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>OR4G4P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromsome  start    end         ensembl id score strand   gene_type  \\\n",
       "0      chr1  11868  14412  ENSG00000223972.4     .      +  pseudogene   \n",
       "1      chr1  14362  29806  ENSG00000227232.4     .      -  pseudogene   \n",
       "2      chr1  29553  31109  ENSG00000243485.2     .      +     lincRNA   \n",
       "3      chr1  34553  36081  ENSG00000237613.2     .      -     lincRNA   \n",
       "4      chr1  52472  54936  ENSG00000268020.2     .      +  pseudogene   \n",
       "\n",
       "  gene_status   gene_name  \n",
       "0       KNOWN     DDX11L1  \n",
       "1       KNOWN      WASH7P  \n",
       "2       NOVEL  MIR1302-11  \n",
       "3       KNOWN     FAM138A  \n",
       "4       KNOWN      OR4G4P  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed = pd.read_csv(r'E:\\work\\mRNA-seq\\annotation\\GRCh37.bed', sep='\\t')\n",
    "biomart = pd.read_csv(r'E:\\work\\mRNA-seq\\annotation\\BiomaRt.GRCh37.p13.anno.gz.txt.pared.tsv', sep='\\t')\n",
    "\n",
    "bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>RefSeq ncRNA ID</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SLC25A26</td>\n",
       "      <td>ENSG00000261657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>solute carrier family 25 (S-adenosylmethionine...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000223116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HMGA1P6</td>\n",
       "      <td>ENSG00000233440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>high mobility group AT-hook 1 pseudogene 6 [So...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RNY3P4</td>\n",
       "      <td>ENSG00000207157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNA, Ro-associated Y3 pseudogene 4 [Source:HGN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LINC00362</td>\n",
       "      <td>ENSG00000229483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>long intergenic non-protein coding RNA 362 [So...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  HGNC symbol   Gene stable ID RefSeq ncRNA ID  \\\n",
       "0    SLC25A26  ENSG00000261657             NaN   \n",
       "1         NaN  ENSG00000223116             NaN   \n",
       "2     HMGA1P6  ENSG00000233440             NaN   \n",
       "3      RNY3P4  ENSG00000207157             NaN   \n",
       "4   LINC00362  ENSG00000229483             NaN   \n",
       "\n",
       "                                         description  \n",
       "0  solute carrier family 25 (S-adenosylmethionine...  \n",
       "1                                                NaN  \n",
       "2  high mobility group AT-hook 1 pseudogene 6 [So...  \n",
       "3  RNA, Ro-associated Y3 pseudogene 4 [Source:HGN...  \n",
       "4  long intergenic non-protein coding RNA 362 [So...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biomart.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromsome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ensembl id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Gene stable ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11868</td>\n",
       "      <td>14412</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14362</td>\n",
       "      <td>29806</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29553</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34553</td>\n",
       "      <td>36081</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52472</td>\n",
       "      <td>54936</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromsome  start    end         ensembl id score strand   gene_type  \\\n",
       "0      chr1  11868  14412  ENSG00000223972.4     .      +  pseudogene   \n",
       "1      chr1  14362  29806  ENSG00000227232.4     .      -  pseudogene   \n",
       "2      chr1  29553  31109  ENSG00000243485.2     .      +     lincRNA   \n",
       "3      chr1  34553  36081  ENSG00000237613.2     .      -     lincRNA   \n",
       "4      chr1  52472  54936  ENSG00000268020.2     .      +  pseudogene   \n",
       "\n",
       "  gene_status   gene_name   Gene stable ID  \n",
       "0       KNOWN     DDX11L1  ENSG00000223972  \n",
       "1       KNOWN      WASH7P  ENSG00000227232  \n",
       "2       NOVEL  MIR1302-11  ENSG00000243485  \n",
       "3       KNOWN     FAM138A  ENSG00000237613  \n",
       "4       KNOWN      OR4G4P  ENSG00000268020  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed[\"Gene stable ID\"] = bed[\"ensembl id\"].str.split(\".\").str[0]\n",
    "\n",
    "bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(57820, 10)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(63677, 4)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biomart.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromsome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ensembl id</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>RefSeq ncRNA ID</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11868</td>\n",
       "      <td>14412</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>NR_046018NR_051986</td>\n",
       "      <td>DEAD/H (Asp-Glu-Ala-Asp/His) box helicase 11 l...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14362</td>\n",
       "      <td>29806</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>NR_024540</td>\n",
       "      <td>WAS protein family homolog 7 pseudogene [Sourc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29553</td>\n",
       "      <td>31109</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NR_036267;NR_036268;NR_036266;NR_036051;NR_036...</td>\n",
       "      <td>microRNA 1302-10 [Source:HGNC Symbol;Acc:38233]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34553</td>\n",
       "      <td>36081</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>NR_026822NR_026818;NR_026820</td>\n",
       "      <td>family with sequence similarity 138, member A ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>52472</td>\n",
       "      <td>54936</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>ENSG00000268020</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>olfactory receptor, family 4, subfamily G, mem...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromsome  start    end         ensembl id score strand   gene_type  \\\n",
       "0      chr1  11868  14412  ENSG00000223972.4     .      +  pseudogene   \n",
       "1      chr1  14362  29806  ENSG00000227232.4     .      -  pseudogene   \n",
       "2      chr1  29553  31109  ENSG00000243485.2     .      +     lincRNA   \n",
       "3      chr1  34553  36081  ENSG00000237613.2     .      -     lincRNA   \n",
       "4      chr1  52472  54936  ENSG00000268020.2     .      +  pseudogene   \n",
       "\n",
       "  gene_status   gene_name   Gene stable ID HGNC symbol  \\\n",
       "0       KNOWN     DDX11L1  ENSG00000223972     DDX11L1   \n",
       "1       KNOWN      WASH7P  ENSG00000227232      WASH7P   \n",
       "2       NOVEL  MIR1302-11  ENSG00000243485  MIR1302-11   \n",
       "3       KNOWN     FAM138A  ENSG00000237613     FAM138A   \n",
       "4       KNOWN      OR4G4P  ENSG00000268020      OR4G4P   \n",
       "\n",
       "                                     RefSeq ncRNA ID  \\\n",
       "0                                 NR_046018NR_051986   \n",
       "1                                          NR_024540   \n",
       "2  NR_036267;NR_036268;NR_036266;NR_036051;NR_036...   \n",
       "3                       NR_026822NR_026818;NR_026820   \n",
       "4                                                NaN   \n",
       "\n",
       "                                         description  \n",
       "0  DEAD/H (Asp-Glu-Ala-Asp/His) box helicase 11 l...  \n",
       "1  WAS protein family homolog 7 pseudogene [Sourc...  \n",
       "2    microRNA 1302-10 [Source:HGNC Symbol;Acc:38233]  \n",
       "3  family with sequence similarity 138, member A ...  \n",
       "4  olfactory receptor, family 4, subfamily G, mem...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(bed, biomart, left_on='Gene stable ID', right_on='Gene stable ID', how='left')\n",
    "\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ensembl id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>RefSeq ncRNA ID</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>NR_046018NR_051986</td>\n",
       "      <td>DEAD/H (Asp-Glu-Ala-Asp/His) box helicase 11 l...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WASH7P</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>NR_024540</td>\n",
       "      <td>WAS protein family homolog 7 pseudogene [Sourc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NR_036267;NR_036268;NR_036266;NR_036051;NR_036...</td>\n",
       "      <td>microRNA 1302-10 [Source:HGNC Symbol;Acc:38233]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAM138A</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>NR_026822NR_026818;NR_026820</td>\n",
       "      <td>family with sequence similarity 138, member A ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>olfactory receptor, family 4, subfamily G, mem...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  HGNC symbol   gene_name         ensembl id   gene_type   gene_name  \\\n",
       "0     DDX11L1     DDX11L1  ENSG00000223972.4  pseudogene     DDX11L1   \n",
       "1      WASH7P      WASH7P  ENSG00000227232.4  pseudogene      WASH7P   \n",
       "2  MIR1302-11  MIR1302-11  ENSG00000243485.2     lincRNA  MIR1302-11   \n",
       "3     FAM138A     FAM138A  ENSG00000237613.2     lincRNA     FAM138A   \n",
       "4      OR4G4P      OR4G4P  ENSG00000268020.2  pseudogene      OR4G4P   \n",
       "\n",
       "                                     RefSeq ncRNA ID  \\\n",
       "0                                 NR_046018NR_051986   \n",
       "1                                          NR_024540   \n",
       "2  NR_036267;NR_036268;NR_036266;NR_036051;NR_036...   \n",
       "3                       NR_026822NR_026818;NR_026820   \n",
       "4                                                NaN   \n",
       "\n",
       "                                         description  \n",
       "0  DEAD/H (Asp-Glu-Ala-Asp/His) box helicase 11 l...  \n",
       "1  WAS protein family homolog 7 pseudogene [Sourc...  \n",
       "2    microRNA 1302-10 [Source:HGNC Symbol;Acc:38233]  \n",
       "3  family with sequence similarity 138, member A ...  \n",
       "4  olfactory receptor, family 4, subfamily G, mem...  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = merged_df[['HGNC symbol','gene_name',\"ensembl id\", 'gene_type', 'gene_name', 'RefSeq ncRNA ID', 'description']]\n",
    "\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_csv(r\"E:\\work\\mRNA-seq\\annotation\\annotation.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
